### PR TITLE
Add zones redis signalr

### PR DESCRIPTION
**Pull request by lowflying**
### JIRA link (if applicable) ###

N/A

### Change description ###

New signal R and redis private dns zone - required for endpoint dns registration

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by timja — CHANGES_REQUESTED** [review]
Do not link sds vNets here, it won't link new ones as they are created, link them in the SDS repo

https://github.com/hmcts/aks-sds-deploy/blob/78c21e61d21a0b146593e1edd5ce5bd7b610bf50/components/01-network/11-inputs-default.tf#L9

File: components/privatelink/signalr-private-link.tf [hunk 1]
```diff
@@ -0,0 +1,13 @@
+data "local_file" "blob-private-link" {
+  filename = "${path.cwd}/../../environments/privatelink/signalr-private-link.yml"
+}
+
+module "blob-private-link" {
```
Comment (timja) [inline components/privatelink/signalr-private-link.tf:5]:
```suggestion
module "signalr-private-link" {
```

File: components/privatelink/signalr-private-link.tf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+data "local_file" "blob-private-link" {
```
Comment (timja) [inline components/privatelink/signalr-private-link.tf:1]:
```suggestion
data "local_file" "signalr-private-link" {
```

File: components/privatelink/signalr-private-link.tf [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  filename = "${path.cwd}/../../environments/privatelink/signalr-private-link.yml"
+}
+
+module "blob-private-link" {
+  source              = "../../modules/azure-private-dns/"
+  cname_records       = yamldecode(data.local_file.blob-private-link.content).cname
+  a_recordsets        = yamldecode(data.local_file.blob-private-link.content).A
+  zone_name           = yamldecode(data.local_file.blob-private-link.content).name
+  vnet_links          = yamldecode(data.local_file.blob-private-link.content).vnet_links
```
Comment (timja) [inline components/privatelink/signalr-private-link.tf:10]: needs updating to signalr
```suggestion
  cname_records       = yamldecode(data.local_file.signalr-private-link.content).cname
  a_recordsets        = yamldecode(data.local_file.signalr-private-link.content).A
  zone_name           = yamldecode(data.local_file.signalr-private-link.content).name
  vnet_links          = yamldecode(data.local_file.signalr-private-link.content).vnet_links
```

**Review by timja — CHANGES_REQUESTED** [review]

**Comment by lowflying** [comment]
> Do not link sds vNets here, it won't link new ones as they are created, link them in the SDS repo
> 
> https://github.com/hmcts/aks-sds-deploy/blob/78c21e61d21a0b146593e1edd5ce5bd7b610bf50/components/01-network/11-inputs-default.tf#L9

Thanks! I was wondering why none of the other had them here: https://github.com/hmcts/aks-sds-deploy/pull/132

**Review by timja — APPROVED** [review]
I suspect you may want the VPN linked?

https://github.com/hmcts/azure-private-dns/blob/13c1194b674226e2962700754b0d6268a272f514/environments/privatelink/postgres-private-link.yml#L18-L19

**Pull request merged**